client/src/app/shared/services/broker.service.ts

```diff
@@ -17,43 +18,55 @@ export class BrokerService {
   constructor(private http: HttpClient) {
   }
 
-  private handleError(operation: any) {
-    return (err: any) => {
-      let errMsg = `error in ${operation}()`;
-      console.log(`${errMsg}:`, err);
-
-      let httpError = {
-        message: "An error occurred in uploading spreadsheet",
-        details: `${err.name}: ${err.message}`
-      };
-
-      let error = err.error && err.error.message && err.error.details ? err.error : httpError;
-      return throwError(error);
-    }
-  }
-
-  public uploadSpreadsheet(formData, isUpdate=false): Observable<UploadResults> {
-    let uploadApiSuffix = isUpdate ? '_update': '';
+  uploadSpreadsheet(formData, isUpdate = false): Observable<UploadResults> {
+    const uploadApiSuffix = isUpdate ? '_update' : '';
     return this.http.post<UploadResults>(`${this.API_URL}/api_upload${uploadApiSuffix}`, formData)
       .pipe(
         tap(data => console.log('server data:', data)),
         catchError(this.handleError('uploadSpreadsheet'))
       );
   }
 
-  public downloadSpreadsheet(submissionUuid): Observable<any> {
+  downloadSpreadsheet(submissionUuid): Observable<any> {
     return this.http
       .get(`${this.API_URL}/submissions/${submissionUuid}/spreadsheet`,
-        { observe: 'response',responseType: 'blob' }
+        {observe: 'response', responseType: 'blob'}
       ).map((res) => {
-        let contentDisposition = res.headers.get('content-disposition');
-        let filename = contentDisposition.split(';')[1].split('filename')[1].split('=')[1].trim();
-        let response = {
-          "data": res.body,
-          "filename": filename
+        const contentDisposition = res.headers.get('content-disposition');
+        const filename = contentDisposition.split(';')[1].split('filename')[1].split('=')[1].trim();
+        const response = {
+          'data': res.body,
+          'filename': filename
         };
-        return response
+        return response;
       });
   }
 
+  generateTemplate(spec: object): Observable<TemplateGenerationResponse> {
+    console.log('requesting')
+    const url = `${this.API_URL}/spreadsheets`;
+    return this.http.post(url, spec)
```

Comment:
The data in the POST request should be..

```
{
   "filename": <desired filename for the .xlsx file>,
   "spec": <the spreadsheet spec object>
}
```

Reply:
yeah, I couldn't determine the type of spec, since the `TemplateSpecification.convert` isn't working yet.  I'm not sure if that'll be easily converted to json as well. The param obj still being mocked in `TemplateGeneratorService._requestToGenerate`.

Reply:
Ahh I see, but what I mean is that the filename has to be specified alongside the spec returned from `TemplateSpecification.convert()`.

Reply:
Added `TemplateGenerationRequestParam`